base/deprecated.jl

```diff
@@ -148,8 +148,6 @@ end
 @deprecate inf{T<:AbstractFloat}(::Type{T})  convert(T,Inf)
 @deprecate nan{T<:AbstractFloat}(::Type{T})  convert(T,NaN)
 
-@deprecate_binding String AbstractString
```

Comment:
Yes, most cases where someone was previously using `String` to mean `AbstractString` will just work, which is why this is sneakily ok.